readthedocs/proxito/middleware.py

```diff
@@ -176,7 +177,10 @@ def process_request(self, request):  # noqa
 
         if '//' in request.path:
             # Remove multiple slashes from URL's
-            return redirect(re.sub('//+', '/', request.get_full_path()))
+            url_parsed = urlparse(request.get_full_path())
+            clean_path = re.sub('//+', '/', url_parsed.path)
+            new_parsed = url_parsed._replace(path=clean_path)
+            return redirect(urlunparse(new_parsed))
```

Comment:
think you can use new_parsed.geturl()

Reply:
We're using it the same way in other places in the code, so I'll keep it for now. We can update it all later.

Reply:
Hrm seems like that is a special case, and we're using `geturl()` too. Will update 👍